packages/cli-kit/src/store.ts

```diff
@@ -40,17 +41,66 @@ const schema = {
 
 let _instance: CLIKitStore | undefined
 
-export function cliKitStore() {
+export async function cliKitStore() {
   if (!_instance) {
+    // eslint-disable-next-line require-atomic-updates
     _instance = new CLIKitStore({
       schema,
       migrations,
       projectName: 'shopify-cli-kit',
+      projectVersion: await constants.versions.cliKit(),
     })
   }
   return _instance
 }
 
+export async function getAppInfo(directory: string): Promise<CachedAppInfo | undefined> {
```

Comment:
This made me think that we should consider breaking store down into smaller stores (i.e. app, theme, hydrogen, cli-kit). Otherwise the cli-kit store will have knowledge on its dependent packages.

Reply:
Yes, I thought about that too, but I didn't want to go any further with the refactor. I'll create a new issue.